compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/BIRGen.java

```diff
@@ -664,7 +664,14 @@ public void visit(BLangFunction astFunc) {
         birFunc.argsCount = astFunc.requiredParams.size()
                 + (astFunc.restParam != null ? 1 : 0) + astFunc.paramClosureMap.size();
         if (astFunc.flagSet.contains(Flag.ATTACHED) && typeDefs.containsKey(astFunc.receiver.getBType().tsymbol)) {
-            typeDefs.get(astFunc.receiver.getBType().tsymbol).attachedFuncs.add(birFunc);
+            BType receiverType = astFunc.receiver.getBType();
+            if (receiverType.tag == TypeTags.OBJECT && ((BObjectType) receiverType).immutableType != null) {
+                BTypeSymbol typeSymbol = ((BObjectType) receiverType).immutableType.effectiveType.tsymbol;
+                if (typeDefs.containsKey(typeSymbol)) {
+                    typeDefs.get(typeSymbol).attachedFuncs.add(birFunc);
```

Comment:
I'm not sure if this is the correct fix. Shouldn't the fix be to add the functions for the intersection type to org.wso2.ballerinalang.compiler.tree.BLangPackage#functions?

Reply:
The function is anyway available in the BLangPackage#functions. The issue is that the function does not get populated for the BIRTypeDefinition instance for the Intersection type

Reply:
Had an offline chat, seems we have to duplicate the function for the Intersection type. Will do the necessary changes